### PR TITLE
Make dev container run in detached mode

### PR DESCRIPTION
**Pull request by serraramiro1**
### Proposed changes

When developing, it's convenient to preserve the container status to keep the builds hot and command history present.
This PR adds such functionality to the `docker/run.sh` script.

#### Type of change

- [ ] 🐛 Bugfix (change which fixes an issue)
- [x] 🚀 Feature (change which adds functionality)
- [ ] 📚 Documentation (change which fixes or extends documentation)

### Checklist

_Put an `x` in the boxes that apply. This is simply a reminder of what we will require before merging your code._

- [x] Lint and unit tests (if any) pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [x] All commits have been signed for [DCO](https://developercertificate.org/)

### CODE REVIEW

File: docker/run.sh [hunk 1]
```diff
@@ -66,10 +66,23 @@ while [[ "$1" != "" ]]; do
     esac
 done
 
+DEV_CONTAINER_NAME=beluga_dev
+
 # Note: The `--build` flag was added to docker compose run after
 # https://github.com/docker/compose/releases/tag/v2.13.0.
 # We have this for convenience and compatibility with previous versions.
 # Otherwise, we could just forward the script arguments to the run verb.
```
Comment (nahueespinosa) [inline docker/run.sh:74]: Let's remove this as it no longer applies.
```suggestion
```

Comment (serraramiro1) [reply docker/run.sh]: Done

File: docker/run.sh [hunk 2]
```diff
@@ -66,10 +66,18 @@ while [[ "$1" != "" ]]; do
...
+# Check if the container exists
+if docker ps -a --format '{{.Names}}' | grep -q $DEV_CONTAINER_NAME; then
+    # Attach to the container if it is running
+    echo "Found dev container running, attaching to it..."
+else
+    echo "Container '$DEV_CONTAINER_NAME' does not exist or is not running, proceeding to run it."
+    PRIVILEGED_CONTAINER=$PRIVILEGED_CONTAINER USERID=$(id -u) GROUPID=$(id -g) docker compose run --rm --detach --name $DEV_CONTAINER_NAME  dev
+fi
+
+USERID=$(id -u) GROUPID=$(id -g) docker exec -it $DEV_CONTAINER_NAME fixuid -q /bin/bash -c 'source /ros_entrypoint.sh && /bin/bash'
```
Comment (nahueespinosa) [inline docker/run.sh:83]: Do we need to run fixuid after the container has been created?

File: docker/run.sh [hunk 3]
```diff
@@ -66,10 +66,18 @@ while [[ "$1" != "" ]]; do
...
+# Check if the container exists
+if docker ps -a --format '{{.Names}}' | grep -q $DEV_CONTAINER_NAME; then
+    # Attach to the container if it is running
+    echo "Found dev container running, attaching to it..."
+else
+    echo "Container '$DEV_CONTAINER_NAME' does not exist or is not running, proceeding to run it."
+    PRIVILEGED_CONTAINER=$PRIVILEGED_CONTAINER USERID=$(id -u) GROUPID=$(id -g) docker compose run --rm --detach --name $DEV_CONTAINER_NAME  dev
+fi
+
+USERID=$(id -u) GROUPID=$(id -g) docker exec -it $DEV_CONTAINER_NAME /bin/bash -c 'source /ros_entrypoint.sh && /bin/bash'
```
Comment (nahueespinosa) [inline docker/run.sh:83]: Last request:
```suggestion
USERID=$(id -u) GROUPID=$(id -g) docker exec -it $DEV_CONTAINER_NAME /ros_entrypoint.sh /bin/bash
```

Comment (nahueespinosa) [reply docker/run.sh:83]: USERID and GROUPID are not used I believe, and we can avoid creating a sub-shell.

**Review by nahueespinosa — APPROVED** [review]
LGTM!

**Pull request merged**